FlowCrypt/Controllers/SideMenu/Main/SideMenuNavigationController.swift

```diff
@@ -96,6 +103,167 @@ final class SideMenuNavigationController: ENSideMenuNavigationController {
     }
 }
 
+// MARK: Refresh keys from EKM
+extension SideMenuNavigationController {
+    private func refreshKeysFromEKMIfNeeded(context: AppContextWithUser) {
+        Task {
+            do {
+                let configuration = try await context.clientConfigurationService.configuration
+                guard configuration.checkUsesEKM() == .usesEKM else {
+                    return
+                }
+                let emailKeyManagerApi = EmailKeyManagerApi(clientConfiguration: configuration)
+                let idToken = try await IdTokenUtils.getIdToken(userEmail: context.user.email)
+                let keydDetails = try await emailKeyManagerApi.getPrivateKeys(idToken: idToken)
+                let localKeys = try context.encryptedStorage.getKeypairs(by: context.user.email)
+                var savedPassPhrase = localKeys.first(where: { $0.passphrase != nil })?.passphrase
+                var isKeyUpdated = false
+                for keyDetail in keydDetails {
+                    guard let savedLocalKey = localKeys.first(where: { $0.primaryFingerprint == keyDetail.primaryFingerprint }) else {
+                        // No keys found in local. Add it
+                        savedPassPhrase = try await saveKeyToLocal(
+                            context: context,
+                            keyDetail: keyDetail,
+                            passPhrase: savedPassPhrase,
+                            isNewKey: true
+                        )
+                        isKeyUpdated = true
+                        continue
+                    }
+                    guard let lastModified = keyDetail.lastModified else {
+                        continue
+                    }
+                    // Key exists in local. Check if saved key is outdated by checking lastModified and update if needed
+                    if savedLocalKey.lastModified ?? 0 < lastModified {
```

Comment:
Let's treat both of them consistently. Therefore I guess `savedLocalKey.lastModified ?? 0 < keyDetail.lastModified ?? 0`

Reply:
I added `guard` above because there might some cases which lastModified is null but want to add/update keys to local